plugins/cilium-cni/main.go

```diff
@@ -712,3 +712,171 @@ func cmdDel(args *skel.CmdArgs) error {
+	}
+
+	cniArgs := types.ArgsSpec{}
+	if err = cniTypes.LoadArgs(args.Args, &cniArgs); err != nil {
+		return cniTypes.NewError(cniTypes.ErrInvalidNetworkConfig, "InvalidArgs",
+			fmt.Sprintf("unable to extract CNI arguments: %s", err))
+	}
+	logger.Debugf("CNI Args: %#v", cniArgs)
+
+	c, err := client.NewDefaultClientWithTimeout(defaults.ClientConnectTimeout)
+	if err != nil {
+		// use ErrTryAgainLater to tell the runtime that this is not a check failure
+		return cniTypes.NewError(cniTypes.ErrTryAgainLater, "DaemonDown",
+			fmt.Sprintf("unable to connect to Cilium daemon: %s", client.Hint(err)))
+	}
+
+	// If this is a chained plugin, then "delegate" to the special chaining mode and be done
+	// Note: CHECK always has PrevResult set, so that doesn't tell us if we're chained.
+	if n.Name != chainingapi.DefaultConfigName {
+		if chainAction := chainingapi.Lookup(n.Name); chainAction != nil {
+			var (
+				ctx = chainingapi.PluginContext{
+					Logger:  logger,
+					Args:    args,
+					CniArgs: cniArgs,
+					NetConf: n,
+					Client:  c,
+				}
+			)
+
+			// err is nil on success
+			err := chainAction.Check(context.TODO(), ctx)
+			logger.Debugf("Chained CHECK %s returned %s", n.Name, err)
+			return err
+
+		} else {
+			logger.Warnf("Unknown CNI chaining configuration name '%s'", n.Name)
+		}
+	}
+
+	// mechanical: parse PrevResult
+	if err := cniVersion.ParsePrevResult(&n.NetConf); err != nil {
+		return err
+	}
+	prevResult, err := cniTypesV1.NewResultFromResult(n.NetConf.PrevResult)
+	if err != nil {
+		return err
+	}
+
+	netNs, err := ns.GetNS(args.Netns)
+	if err != nil {
+		return cniTypes.NewError(cniTypes.ErrInvalidEnvironmentVariables, "NoNetNS",
+			fmt.Sprintf("failed to open netns %q: %s", args.Netns, err))
+	}
+	defer netNs.Close()
+
+	// Ask the agent for the endpoint's health
+	eID := fmt.Sprintf("container-id:%s", args.ContainerID)
+	logger.Debugf("Asking agent for healthz for %s", eID)
+	epHealth, err := c.EndpointHealthGet(eID)
+	if err != nil {
+		return cniTypes.NewError(types.CniErrHealthzGet, "HealthzFailed",
+			fmt.Sprintf("failed to retrieve container health: %s", err))
+	}
+
+	if epHealth.OverallHealth == models.EndpointHealthStatusFailure {
+		return cniTypes.NewError(types.CniErrUnhealthy, "Unhealthy",
+			"container is unhealthy in agent")
+	}
+	logger.Debugf("Container %s has a healthy agent endpoint", args.ContainerID)
+
+	// Verify that the interface exists and has the desired IP address
+	// we can get the IP from the CNI previous result.
+	if err := verifyInterface(netNs, args.IfName, prevResult); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// verifyInterface verifies that a given interface exists in the netns
+// with the given addresses
+func verifyInterface(netns ns.NetNS, ifName string, expected *cniTypesV1.Result) error {
+	wantAddresses := []*cniTypesV1.IPConfig{}
+	for idx, iface := range expected.Interfaces {
+		if iface.Sandbox == "" {
```

Comment:
Question: Under what circumstance is this empty?

Reply:
Great question. the CNI Result type allows you to add non-sandbox interfaces if they are relevant (e.g. the bridge on the host). The way to signify this is that the Sandbox field is empty.